scrnaseq.Rmd

```diff
@@ -285,36 +364,60 @@ switch (param$debugging_mode,
 # Set output hooks
 knitr::knit_hooks$set(message=format_message, warning=format_warning)
 
-# Create output directories <==== How about we organise a bit more :-)
+# Create output directories
 if (!file.exists(param$path_out)) dir.create(param$path_out, recursive=TRUE, showWarnings=FALSE)
-#dir.create(paste(param$path_out, "images", sep="/"), recursive=TRUE, showWarnings=FALSE)
-#dir.create(paste(param$path_out, "marker", sep="/"), recursive=TRUE, showWarnings=FALSE)
-#dir.create(paste(param$path_out, "degs", sep="/"), recursive=TRUE, showWarnings=FALSE)
-output_files = list()
+dir.create(file.path(param$path_out, "figures"), recursive=TRUE, showWarnings=FALSE)
+dir.create(file.path(param$path_out, "marker_degs"), recursive=TRUE, showWarnings=FALSE)
+dir.create(file.path(param$path_out, "annotation"), recursive=TRUE, showWarnings=FALSE)
+dir.create(file.path(param$path_out, "data"), recursive=TRUE, showWarnings=FALSE)
+dir.create(file.path(param$path_out, "export"), recursive=TRUE, showWarnings=FALSE)
+
+# Path for figures in png and pdf format (trailing '/' is needed)
+knitr::opts_chunk$set(fig.path=paste0(file.path(param$path_out, "figures"), "/"))
+
+# Path for annotation
+if (is.null(param$file_annot)) param$file_annot = file.path(param$path_out, "annotation", paste0(param$mart_dataset, ".v", param$annot_version, ".annot.txt"))
 
 # Do checks
 error_messages = c()
 
+# Check parameters and parse entries (e.g. numbers) so that they are valid
+param = check_parameters_scrnaseq(param)
+error_messages = c(error_messages, param[["error_messages"]])
 # Check installed packages
-error_messages = c(error_messages, check_installed_packages())
+error_messages = c(error_messages, check_installed_packages_scrnaseq())
 # Check python
 error_messages = c(error_messages, check_python())
-# Check parameters
-error_messages = c(error_messages, check_parameters(param))
+# Check pandoc
+error_messages = c(error_messages, check_pandoc())
 # Check enrichR
 error_messages = c(error_messages, check_enrichr(param$enrichr_dbs))
 # Check ensembl
 error_messages = c(error_messages, check_ensembl(biomart="ensembl", 
                                                  dataset=param$mart_dataset, 
                                                  mirror=param$biomart_mirror, 
                                                  version=param$annot_version,
-                                                 attributes=param$mart_attributes))
+                                                 attributes=param$mart_attributes,
+                                                 file_annot=param$file_annot))
 
 # Stop here if there are errors
 if (length(error_messages)) stop(paste(c("", paste("*", error_messages)), collapse="\n"))
 ```
 
 # Read data
+The workflow works with pre-processed 10x data, as well as pre-processed SmartSeq-2 and other data that are represented by a simple table for counts per gene and cell. 
```

Comment:
Take out for now, until we add all explanatory texts.

Reply:
Removed